src/lib/Builder.hs

```diff
@@ -1165,13 +1165,13 @@ projMethod methodName dict = do
       emitOp $ ProjMethod dict methodIdx
     _ -> error $ "Not a dict: " ++ pprint dict
 
-intToIndex :: forall m n. (Builder m, Emits n) => IxType n -> Atom n -> m n (Atom n)
-intToIndex (IxType _ dict) i = do
+natToIndex :: forall m n. (Builder m, Emits n) => IxType n -> Atom n -> m n (Atom n)
```

Comment:
Can we actually rename those to match their user-level names? I've always found those confusing.